ui/dashboards/src/components/Panel/Panel.test.tsx

```diff
@@ -41,23 +45,36 @@ describe('Panel', () => {
         options: {},
       },
     };
+
+    initialState = {
+      isEditMode: false,
+      dashboardSpec: testDashboard.spec,
+    };
   });
 
   // Helper to render the panel with some context set
-  const renderPanel = () => {
+  const renderPanel = (initialState: DashboardStoreProp) => {
     const { addMockPlugin, pluginRegistryProps } = mockPluginRegistryProps();
     addMockPlugin(FAKE_PANEL_PLUGIN);
 
     renderWithContext(
-      <PluginRegistry {...pluginRegistryProps}>
-        <Panel {...props} />
-      </PluginRegistry>
+      <DashboardProvider initialState={initialState}>
+        <PluginRegistry {...pluginRegistryProps}>
+          <Panel {...props} />
+        </PluginRegistry>
+      </DashboardProvider>
     );
   };
 
   it('should render name and info icon', async () => {
-    renderPanel();
+    renderPanel(initialState);
     await screen.findByText('Fake Panel');
     screen.queryByLabelText('info-tooltip');
   });
+
+  it('should render edit panel icon when in edit mode', async () => {
+    initialState.isEditMode = true;
+    renderPanel(initialState);
+    await screen.getByTestId('PencilIcon');
```

Comment:
will update this test to not use testId when we implement the edit panel feature